Face/FindSimilar/Program.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Azure.CognitiveServices.Vision.Face;
+using Microsoft.Azure.CognitiveServices.Vision.Face.Models;
+
+namespace FindSimilar
+{
+    public class Program
+    {
+        // Create a client.
+        public static string ApiKey = "ENTER YOUR KEY HERE";
+
+        public static IFaceClient Client = new FaceClient(new ApiKeyServiceClientCredentials(ApiKey))
+        {
+            Endpoint = "ENTER YOUR ENDPOINT HERE"
+        };
+
+        public static void Main(string[] args)
+        {
+            FindSimilar_in_FaceIds();
+            FindSimilar_in_FaceList();
+            FindSimilar_in_LargeFaceList();
+            Console.ReadLine();
+        }
+
+        public static void FindSimilar_in_FaceIds()
+        {
+            List<string> imageFileNames =
+                new List<string> { "Family1-Dad1.jpg", "Family1-Dad2.jpg", "Family1-Son1.jpg" };
+            IList<Guid?> faceIds = new List<Guid?>();
+
+            foreach (var imageFileName in imageFileNames)
+            {
+                // Read image file.
+                using (FileStream stream = new FileStream(Path.Combine("Images", imageFileName), FileMode.Open))
+                {
+                    // Detect faces from image stream.
+                    IList<DetectedFace> detectedFaces = Client.Face.DetectWithStreamAsync(stream).Result;
+                    if (detectedFaces == null || detectedFaces.Count == 0)
+                    {
+                        Console.WriteLine($"[Error] No face detected from image `{imageFileName}`.");
+                        return;
+                    }
+
+                    Console.WriteLine($"{detectedFaces.Count} faces detected from image `{imageFileName}`.");
+                    if (detectedFaces[0].FaceId == null)
+                    {
+                        Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for verification purpose.");
+                        return;
+                    }
+
+                    // Add detected faceId to faceIds.
+                    faceIds.Add(detectedFaces[0].FaceId);
+                }
+            }
+
+            // Find similar example for find similar faces from faceIds with similar confidence.
+            IList<SimilarFace> similarResults = Client.Face.FindSimilarAsync(faceIds[0].Value, null, null, faceIds).Result;
+            foreach (var similarResult in similarResults)
```

Comment:
Let's use 0.5 as threshold to the confidence to distinguish whether same person or not.
1. Same means confidence  > 0.5
2. Positive means they are actually same person (ground truth)

Reply:
done